recOrder/acq/acquisition_workers.py

```diff
@@ -345,44 +344,22 @@ def _save_imgs(self, phase, meta=None):
         -------
 
         """
-
-        writer = WaveorderWriter(self.snap_dir)
-
-        # initialize
-        chunk_size = (1, 1, 1, phase.shape[-2], phase.shape[-1])
         prefix = self.calib_window.save_name
         name = f"PhaseSnap.zarr" if not prefix else f"{prefix}_PhaseSnap.zarr"
 
-        # create zarr root and position group
-        writer.create_zarr_root(name)
-
-        # Check if 2D
-        if len(phase.shape) == 2:
-            writer.init_array(
-                0,
-                (1, 1, 1, phase.shape[-2], phase.shape[-1]),
-                chunk_size,
-                ["Phase2D"],
-            )
-            z = 0
-
-        # Check if 3D
-        else:
-            writer.init_array(
-                0,
-                (1, 1, phase.shape[-3], phase.shape[-2], phase.shape[-1]),
-                chunk_size,
-                ["Phase3D"],
-            )
-
-            z = slice(0, phase.shape[-3])
-
-        # Write data to disk
-        writer.write(phase, p=0, t=0, c=0, z=z)
-
-        current_meta = writer.store.attrs.asdict()
-        current_meta["recOrder"] = meta
-        writer.store.attrs.put(current_meta)
+        with open_ome_zarr(
+            os.path.join(self.snap_dir, name),
+            layout="fov",
+            mode="w",
+            channel_names=["Phase" + str(phase.ndim) + "D"],
+        ) as dataset:
+            dataset["0"] = phase[
+                (5 - phase.ndim) * (np.newaxis,) + (Ellipsis,)
+            ]
+            # TODO WRITE METADATA
+            # current_meta = writer.store.attrs.asdict()
+            # current_meta["recOrder"] = meta
+            # writer.store.attrs.put(current_meta)
```

Comment:
@ziw-liu can you help me write the recOrder metadata into the v0.4 zarr stores?

Reply:
Haven't tested https://github.com/mehta-lab/recOrder/pull/331/commits/969752ed7f1c244cd8e9780ab53b6552ac3f0c38. Can you try it?

Reply:
Thank you! Works great. I'll add it elsewhere.